tests/torchtune/rl/test_parameter_server.py

```diff
@@ -0,0 +1,167 @@
+            for k, v in self.model.state_dict().items():
+                torch.distributed.send(v, 1)
+            ray.get(self.parameter_server.release_state_dict_lock.remote())
+
+    @ray.remote(num_cpus=1, num_gpus=1)
+    class DummyCollector:
+        def __init__(self, weight_receiver, tp_size):
+            os.environ["VLLM_USE_V1"] = "0"
+
+            self.inference_server = LLM(
+                "gpt2",
+                enforce_eager=True,
+                dtype="bfloat16",
+                worker_cls=VLLMWorkerWrapper,
+                tensor_parallel_size=tp_size,
+            )
+
+            self.weight_update_receiver = weight_receiver
+            self.weight_update_receiver.register_collector(self)
+
+        def update_policy_weights_(self) -> None:
+            self.weight_update_receiver()
+
+        def check_weights_changed(self, fill_val) -> bool:
+            return self.inference_server.collective_rpc(
+                "_check_weights_changed", args=(fill_val,)
+            )
+
+    class TestParamServer:
+        def _get_env_vars(self, rank):
+            env_vars = {
+                "RANK": rank,
+                "WORLD_SIZE": 2,
+                "MASTER_ADDR": "127.0.0.1",
+                "MASTER_PORT": "29500",
+            }
+            return env_vars
+
+        def _get_config(self, tp_size: int = 1):
+            config_str = f"""
+vllm:
+    tp_size: {tp_size}
+            """
+            cfg = OmegaConf.create(config_str)
+            return cfg
+
+        @gpu_test(gpu_count=2)
+        def test_receive_from_trainer(self) -> None:
+            ray.init(num_cpus=10, num_gpus=2)
+
+            try:
+                trainer_handle = DummyTrainer.remote(self._get_env_vars(0))
+                cfg = self._get_config()
+                ps_handle = VLLMParameterServer.remote(
+                    cfg, None, None, self._get_env_vars(1)
+                )
+                trainer_handle.register_parameter_server.remote(ps_handle)
+                model_metadata = ray.get(trainer_handle.get_model_metadata.remote())
+                ps_handle.register_model_metadata.remote(model_metadata)
+                ps_weights = ray.get(ps_handle._get_server_weights.remote())
+
+                for k, v in model_metadata.items():
+                    assert ps_weights[k].dtype == v[0]
+                    assert ps_weights[k].shape == v[1]
+                    assert torch.equal(ps_weights[k], torch.zeros_like(ps_weights[k]))
+
+                ray.get(trainer_handle.sync_weights.remote())
+
+                updated_ps_weights = ray.get(ps_handle._get_server_weights.remote())
+                for k, v in model_metadata.items():
+                    assert torch.equal(
+                        updated_ps_weights[k], torch.ones_like(updated_ps_weights[k])
+                    )
+            finally:
+                ray.shutdown()
+
+        @gpu_test(gpu_count=4)
+        @pytest.mark.parametrize("tp_size", (1, 2))
```

Comment:
I personally like [Hypothesis](https://hypothesis.readthedocs.io/en/latest/) to parametrize tests, but anything works for me

Reply:
I'd +1 pytest parametrize, I'm very familiar with that :p